### PR TITLE
Add a `prow_ignored` field for non-Prow fields in job config.

### PR DESCRIPTION
**Pull request by cjwagner**
Addresses part 2 of https://github.com/kubernetes/test-infra/issues/24537
/assign @chaodaiG

### CODE REVIEW

File: prow/config/config.go [hunk 1]
```diff
@@ -115,6 +115,10 @@ type JobConfig struct {
 
 	// DecorateAllJobs determines whether all jobs are decorated by default
 	DecorateAllJobs bool `json:"decorate_all_jobs,omitempty"`
+
+	// ProwIgnored is a well known, unparsed field where non-Prow fields can
+	// be defined without conflicting with unknown field validation.
+	ProwIgnored *json.RawMessage `json:"prow_ignored,omitempty"`
```
Comment (listx) [inline prow/config/config.go:121]: Nit: I would have named it just `Ignored`.

Comment (cjwagner) [reply prow/config/config.go:121]: I think the specificity will be helpful in cases where there are multiple tools consuming the file.

**Review by chaodaiG — APPROVED** [review]
LGTM
/hold
in case you want it to be reviewed by someone else

**Comment by cjwagner** [comment]
/unhold
Rebased to remove the conflicting file. I'm not sure why why the ProwJob CRD was updated by the code gen, but that seems incorrect.

**Review by chaodaiG — APPROVED** [review]

**Pull request merged**